CMake/CheckAndAddFlag.cmake

```diff
@@ -27,13 +27,23 @@ function(check_and_add_flag var flag)
     message(FATAL_ERROR "check_and_add_flag called with incorrect arguments: ${ARGN}")
   endif()
 
+  set(is_c "$<COMPILE_LANGUAGE:C>")
+  set(is_cxx "$<COMPILE_LANGUAGE:CXX>")
+
+  # The Visual Studio generators don't support COMPILE_LANGUAGE
+  # So we fail all the C flags and only actually test CXX ones
+  if(CMAKE_GENERATOR MATCHES "Visual Studio")
```

Comment:
I mean that practically all of the checks against C _and_ C++ could be turned into just a check against C and unconditionally added to `compile_options`. It wouldn't lead to any duplicates.